### PR TITLE
#607 Refactor Mail class

### LINKED ISSUE #607: Refactor the Mail class to adhere to DRY principles
#### Issue Summary

For an example, please see `[addRecipientEmail](https://github.com/sendgrid/sendgrid-php/blob/master/lib/mail/Mail.php#L193)`. That method consolidates much of the functionality of the `addTo`, `addCc` and `addBcc` methods. We should similarly combine functionality where appropriate in this code base as several of the getters/setters share similar functionality.

### PR DESCRIPTION
**Pull request by hulkur**
Resolves #607 

Refactored `addHeader`, `addSubstitution`, `addCustomArg` and `setSendAt` to use common method for personalization handling.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-php?pullRequest=706) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-php?pullRequest=706) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-php?pullRequest=706) it.</sub>

File: lib/mail/Personalization.php [hunk 1]
```diff
@@ -240,6 +240,19 @@ public function setSendAt($send_at)
...
+     * Alias to SendAt()
+     * to have similar naming as other mail parts as used in Mail class
+     *
+     * @param SendAt $send_at SendAt object
+     *
+     * @throws TypeException
+     */
+    public function addSendAt($send_at)
+    {
+        $this->setSendAt($send_at);
```
Comment (peter279k) [inline lib/mail/Personalization.php]: Does the `$send_at` variable should change into `$sendAt` because of the variable declaration consistency.

Comment (hulkur) [reply lib/mail/Personalization.php]: It's named as in `setSendAt` method.

Comment (peter279k) [reply lib/mail/Personalization.php]: I mean the variable declaration should use the `camel-case` style?

Comment (hulkur) [reply lib/mail/Personalization.php]: Maybe. It's consistent with existing code in the class.

Comment (peter279k) [reply lib/mail/Personalization.php]: Ok. I think it's okay to use `$send_at` variable because it's consistent for this `sendAt` method :).

**Comment by thinkingserious** [comment]
Hello @hulkur,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by hulkur** [comment]
@thinkingserious any updates on swag?

Need a shirt! Must have shirt! :)

**Comment by childish-sambino** [comment]
Replaced by https://github.com/sendgrid/sendgrid-php/pull/952

**Pull request closed without merging**